sdk/src/dlob/DLOB.ts

```diff
@@ -1689,4 +1689,66 @@ export class DLOB {
 			yield nodeLists.trigger.below;
 		}
 	}
+
+	public getSpreadInfo(
+		marketIndex: number, 
+		marketType: MarketType,
+		driftClient: DriftClient,
+		slot: number): 
+		{rawSpread: BN; spreadPct: number; } {
+
+		const oraclePriceData =
+				driftClient.getOracleDataForPerpMarket(marketIndex);
+
+		let bestBid;
+		let bestAsk;
+
+		if (isVariant(marketType, 'perp')) {
+			const market = driftClient.getPerpMarketAccount(marketIndex);
+
+			const fallbackAsk = calculateAskPrice(market, oraclePriceData);
+			const fallbackBid = calculateBidPrice(market, oraclePriceData);
+
+			bestAsk = this.getBestAsk(
+				marketIndex,
+				fallbackAsk,
+				slot,
+				marketType,
+				oraclePriceData
+			);
+			bestBid = this.getBestBid(
+				marketIndex,
+				fallbackBid,
+				slot,
+				marketType,
+				oraclePriceData
+			);
+
+		} else if (isVariant(marketType, 'spot')) {
+			bestAsk = this.getBestAsk(
+				marketIndex,
+				undefined,
+				slot,
+				marketType,
+				oraclePriceData
+			);
+			bestBid = this.getBestBid(
+				marketIndex,
+				undefined,
+				slot,
+				marketType,
+				oraclePriceData
+			);
+
+		}
+
+		if (bestBid && bestAsk) {
+			const rawSpread = bestBid.sub(bestAsk).abs();
```

Comment:
if the bid and ask cross, should this be negative?

Reply:
updated